dev/com.ibm.ws.config/src/com/ibm/ws/config/xml/internal/ConfigRefresher.java

```diff
@@ -136,7 +147,13 @@ private synchronized void doRefresh(Map<String, DeltaType> variableDelta) {
             // Let the notification show that we got an error while making the configuration changes
             configUpdatesDelivered.setResult(e);
         } finally {
-            changesEnded(configUpdatesDelivered);
+
+            if (CheckpointPhase.getPhase().restored() && changesEndedHook.runMultiThreaded) {
```

Comment:
The `INACTIVE` returns true from `restored()`.  I am surprised this does not NPE on `changesEndedHook` when the phase is `INACTIVE` ...

But now I see that you would always create the `new ChangesEndedHook();` in the constructor in this case of `INACTIVE` phase.  Is that intended?  It is a bit confusing to me.

I suggest changing the constructor to always create the ChangesEndedHook and check the `INACTIVE` phase:

```java
        changesEndedHook = new ChangesEndedHook();
        if (checkpoint != CheckpointPhase.INACTIVE && !checkpoint.restored()) {
            checkpoint.addMultiThreadedHook(changesEndedHook);
        }
```
Make `changesEndedHook` final.

Reply:
Done.  Yes, that was unintentional.  I had equated !restored() to (!phase.INACTIVE && !phase.restored).  Also, it was impossible to make the hook final in the original solution.